crates/noirc_evaluator/src/ssa_refactor/ssa_builder/mod.rs

```diff
@@ -251,7 +278,12 @@ impl FunctionBuilder {
         self.insert_instruction(Instruction::ArrayGet { array, index }, element_type).first()
     }
 
-    /// Insert an instruction to create a new array with the given index replaced with a new value
+    /// Insert an array_set instruction to the end of the block.
+    ///
+    /// This will create a new array with the given index replaced with a `value`.
+    /// Note: This will not modify `array`. Arrays are immutable in SSA.
+    ///
+    /// Returns the `ValueId` of the newly created array. This will be a reference.
```

Comment:
```suggestion
    /// This will create a new array with the given index replaced with `value`.
    /// Note: This will not modify `array`. Arrays are immutable in SSA.
    ///
    /// Returns the `ValueId` of the newly created array.
```
The returned array has a `Type::Array { ... }` type rather than a reference type